strider/query_planner.py

```diff
@@ -388,21 +400,19 @@ async def generate_plan(
                         **op._asdict(),
                         **kp,
                     })
+            plans.append(plan)
 
-    # No steps in the plan here means we didn't find any permutations where
-    # the plan covers all nodes and edges
-    if len(plan) == 0:
+    if len(plans) == 0:
```

Comment:
We're handling this case in two places, here and [there](https://github.com/ranking-agent/strider/pull/45/files#diff-618b9679423647529785427fae14a18324c4029b0e01ba24802812b3c0b5ebd1R134). We should pick one.

Reply:
This is weird, but I'll try to explain my reasoning. Finding zero plans isn't really something we should throw an exception for from generate_plans because no plans is still a valid result. However, we still want to know what went wrong and there's no way to tell exactly what went wrong when we handle it in `fetcher.py`. That is why we're logging the error in the generate_plans function and then handling it later.

There are some other options on how to do this. We could do a double return value (Go style) where the second return value is the error message. Let me know what you think would be the best fit for the project.

Reply:
I think the use of `logger.error()` should be closely coupled with actually throwing errors and returning non-2xx status codes. My inclination is to move this logging step out to sit beside the `raise`.

Reply:
The problem is that we don't know what error message to log if the logger.error call is in fetcher.py. The error could be "We couldn't find any possible plans starting from a pinned node" or "We couldn't find a KP for every edge in your query graph".

Reply:
Ah, I see the issue. To build on your point about separation of concerns, it should apply to logging, too. If `generate_plan()` is not going to raise an error, it shouldn't be logging one either. We could switch these logs to warnings, return an empty list, then log and raise an error when empty list becomes a terminal problem.

Reply:
That makes sense